pyccel/ast/core.py

```diff
@@ -2707,6 +2706,24 @@ def is_static(self):
         """
         return self._is_static
 
+    @property
+    def is_annotated(self):
+        """
+        Indicates if the function is annotated.
+        """
+        return self._is_annotated
```

Comment:
Does this need to be an attribute? Couldn't you just use the syntactic node directly and define:
```suggestion
    @property
    def is_annotated(self):
        """
        Indicates if the function is annotated.
        """
        return self.pyccel_staging != 'syntactic'
```
?

Reply:
Assuming that "is annotated" means that the semantic stage has been used to annotate all nodes inside the FunctionDef? Or does it mean that all arguments and results have annotations?

Reply:
This is used to check if all nodes inside the FunctionDef have been annotated, in this implementation, we pass through the FunctionDef node multiple times, so I need to check if it has been annotated or not

Reply:
So you create lots of intermediate `FunctionDef` objects? Is it not possible to just return the original syntactic object if you discover that the `FunctionDef` can't be annotated yet?

Reply:
I think the `pyccel_staging` variable is already used in a couple of places to check whether or not objects are annotated. I don't think it is a good idea to have multiple mechanisms for this.

Reply:
doesn't the pyccel_stage change when we enter the semantics parser? because it is a singleton class

Reply:
The `pyccel_stage` which is the instance of `PyccelStage` changes when we enter the semantic parser, but the property `ast.Basic.pyccel_staging` (https://github.com/pyccel/pyccel/blob/e371f27b97024cc1d5632029dc54a96e63023270/pyccel/ast/basic.py#L63) doesn't change as this is simply a string.

Reply:
done in [24de7a6](https://github.com/pyccel/pyccel/pull/1728/commits/24de7a63ea38d2f9c27e1a48c04cb9460108274c)